carelink.js

```diff
@@ -196,15 +196,18 @@ var Client = exports.Client = function (options) {
     }
 
     function doLoginEu4(response, next) {
-        logger.log('GET ' + CARELINKEU_LOGIN4_URL);
+        const regexUrl = /(<form action=")(.*)" method="POST"/gm;
+        let url = regexUrl.exec(response.body)[2] || '';
 
-        let ps = params(response.request.body);
+        logger.log('GET ' + url);
 
+        // Session data is changed, need to get it from the html body form
+        let ps = params(response.request.body);
```

Comment:
I found this confusing, doesn't `params` expect to parse a url?  But this looks like a request.body (html)?  I thought maybe using cheerio might be good here, or conversely `qs.parse` if needed?

Reply:
Agree very messy, changed it to regex. Still not perfect, but for extracting a string I think more than enough.